### PR TITLE
Replace legacy charts with new ones

### PR DESCRIPTION
**Pull request by fpliger**
Replaces old legacy charts left with new ones based on the CompositeGlyphs and more advanced builders design.

### CODE REVIEW

**Comment by fpliger** [comment]
@rothnic do you have more changes that you want to push besides the `Donut` implementation? Otherwise, if it's `ready-ish` I'd propose to make it ready. What do you think?

**Comment by fpliger** [comment]
@rothnic , what do you think about merging this today? I'd leave the Donut (and other tasks) for a following PR as this one is already quite big and it'd be good to merge it soon

**Comment by damianavila** [comment]
@fpliger @rothnic what the status of this one? The failures in TravisCI seems related.

**Comment by fpliger** [comment]
@damianavila yes.. I plan to fix them in soon.

**Comment by fpliger** [comment]
all PR related issues should be fixed. The tests failing are unrelated to the PR and being fixed with #3180 ...

**Comment by bryevdv** [comment]
@fpliger does #3180 need to be merged first?

**Comment by fpliger** [comment]
ok.. so #3190 got it's way to master and should fix this as well.. Will merge master and merge back if tests pass

**Comment by fpliger** [comment]
@bokeh/dev fixed and should ready to be merged to master... There is a weird thing I've noticed left on the scatter where the simple examples are blank. It seems to be related to the fact that when `color`, `line_color` and/or 'fill_color` are not specified the related `glyph` class assigns `None` to the color attribute instead of it's default property value.. Since this PR is quite big and being merged late I'd propose to merge it as is (if there are not other big issues or bad things that emerge from review) and fix it in a following PR.

File: examples/charts/file/timeseries.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+IBM = pd.read_csv(
+    "http://ichart.yahoo.com/table.csv?s=IBM&a=0&b=1&c=2000&d=0&e=1&f=2010",
+    parse_dates=['Date'])
+
+xyvalues = OrderedDict(
+    AAPL=AAPL['Adj Close'],
+    Date=AAPL['Date'],
+    MSFT=MSFT['Adj Close'],
+    IBM=IBM['Adj Close'],
+)
```
Comment (bryevdv) [inline examples/charts/file/timeseries.py]: Does this  need to be an `OrderedDict`? It's probably an unusual data structure for many users. Will a simple `dict` work? (I am not sure anything about the order here matters?) 

Also, if the time series happen to line up, would this work with a `DataFrame` like all the other new examples?

Comment (bryevdv) [inline examples/charts/file/timeseries.py]: Also this example and `examples/charts/file/stocks_timeseries.py` seem to overlap alot? Perhaps they can be merged, or one can be deleted?

File: examples/charts/file/steps.py [hunk 2]
```diff
@@ -1,20 +1,18 @@
...
-#xyvalues = list(xyvalues.values())
-#xyvalues = np.array(list(xyvalues.values()))
-
-output_file("steps.html", title="line.py example")
-
-chart = Step(xyvalues, title="Steps", ylabel='measures', legend='top_left')
-
-show(chart)
+output_file("steps.html")
+show(line)
```
Comment (bryevdv) [inline examples/charts/file/steps.py]: TBH I find this example extremely weird because the synthetic data is really weird. Is there some better example we can construct? Or maybe just any kind of more meaningful title would help.

File: examples/charts/file/timeseries.py (hunk 1 above)
Comment (rothnic) [inline examples/charts/file/timeseries.py]: It could just be a dict I believe. I think I converted other ordereddict examples to just dicts. Using it in this way won't result in the implied order anyways.

File: examples/charts/file/horizon.py [hunk 3]
```diff
@@ -15,17 +15,17 @@
...
-    AAPL=AAPL['Adj Close'],
-    Date=AAPL['Date'],
-    MSFT=MSFT['Adj Close'],
-    IBM=IBM['Adj Close'],
+xyvalues = OrderedDict([
+    ('AAPL', AAPL['Adj Close']),
+    ('Date', AAPL['Date']),
+    ('MSFT', MSFT['Adj Close']),
+    ('IBM', IBM['Adj Close'])]
 )
```
Comment (bryevdv) [inline examples/charts/file/horizon.py]: Same question as below, if `dict` will do, let's change to `dict`.

File: examples/charts/file/area.py [hunk 4]
```diff
@@ -11,11 +14,19 @@
...
+    xlabel='time', ylabel='memory', legend="top_left"
 )
 
-show(area)
+show(
+    vplot(
+        area1,
+        area2
+    )
+)
```
Comment (bryevdv) [inline examples/charts/file/area.py]: This can all go on one line easily:
```
show(vplot(area1, area2))
```
Many of these go or will go in a gallery so vertical space is not unimportant.

File: examples/charts/file/area.py [hunk 5]
```diff
@@ -1,6 +1,9 @@
 from collections import OrderedDict
 
-from bokeh._legacy_charts import Area, show, output_file
+from bokeh.charts import Area, show, vplot, output_file, defaults
+
+defaults.width = 400
+defaults.height = 400
 
 # create some example data
 xyvalues = OrderedDict(
```
Comment (bryevdv) [inline examples/charts/file/area.py]: I really don't like `xyvalues` either. It looks unusual, it has too many syllables. I'd *really* like to call these something else, everywhere. Even just generic `data` would be better.

File: examples/charts/file/dots.py [hunk 6]
```diff
@@ -9,16 +9,19 @@
...
-
 output_file("dots.html")
 
-dots = Dot(
-    xyvalues, cat=['lists','loops','dicts', 'gen exp', 'exceptions'],
-    title="Dots Example", ylabel='Performance', legend=True
-)
+from bokeh.charts import Bar, output_file, show, hplot
+
+# best support is with data in a format that is table-like
```
Comment (bryevdv) [inline examples/charts/file/dots.py]: Only support, right? 
maybe:
```
# data must be table-like
```

File: examples/charts/file/dots.py [hunk 7]
```diff
@@ -9,16 +9,19 @@
     jython=[22, 43, 10, 25, 26],
 )
```
Comment (bryevdv) [inline examples/charts/file/dots.py]: `xyvalues` appears to be unused?

File: examples/charts/file/dots.py (hunk 6 above)
Comment (rothnic) [inline examples/charts/file/dots.py]: Should probably reference the documentation if anything where it describes the accepted input types since table-like is a bit ambiguous, since charts tries to convert any set of inputs into a dataframe internally.

File: examples/charts/file/area.py (hunk 5 above)
Comment (rothnic) [inline examples/charts/file/area.py]: Agree, we should go through and scrub the language across the examples. I have renamed some, but some of the examples still have the xyvalues in there. Some of the examples use names to differentiate the data as needed if the examples script uses more than one source.

Comment (bryevdv) [inline examples/charts/file/area.py]: We can do this work in a separate PR but if so please make an issue to explicitly do a pass over the examples to make them all consistent, concise, non-redundant, etc.

**Comment by fpliger** [comment]
+1 for all comments but would vote for addressing them on a following PR

**Comment by bryevdv** [comment]
@bokeh/core I tested this and the other charts PRs by running the examples. There are definitely issues but I think most or all are related to recent properties changes. The charts PRs are somewhat big and I would live to merge this and #3208 as-is, and add set of issues that catalog all the problems to clean up in smaller follow on PRs. I will merge these in a few hours without objection.
